Update doc (#258)
Sylius/Stack: docs/cookbook/admin_panel/menu.md

    }
}
```

### Opening a link in a new tab

> Target blank directive can be added to link that either are top level menu items without children or children items. In other words this can be configured to item that do not have children.

Reviewer: I think this sentence is redundant with the next one on line 98 - I think we should delete this one entirely up to line 96 and only keep the line 98 one which is much better.

> Target blank directive can be added to link that either are top level menu items without children or children items. In other words this can be configured to item that do not have children.

Reviewer: ```suggestion
```

> Target blank directive can be added to link that either are top level menu items without children or children items. In other words this can be configured to item that do not have children.

Author: I updated the doc like @stlgaits suggested

> This can be configured by setting a link attribute on a menu item:

Reviewer: ```suggestion

```

You can make a link open in a new browser tab by adding the `target="_blank"` link attribute to menu items. This can be applied to top-level menu items without children, or to child items themselves. In other words, it only works for items that do not have submenus.

To configure this behavior, set the target link attribute on the menu item as shown below:

> 

Reviewer: ```suggestion
{% code title="src/Menu/MenuBuilder.php" lineNumbers="true" %}
```

```php
// ...
#[AsDecorator(decorates: 'sylius_admin_ui.knp.menu_builder')]
final readonly class MenuBuilder implements MenuBuilderInterface
{
    // ...

    public function createMenu(array $options): ItemInterface
    {
        $menu = $this->factory->createItem('root');
        // ...

>         $this->aaddApiDocMenuddApiDocMenu($menu);

Reviewer 2: ```suggestion
        $this->addApiDocMenu($menu);
```

Reviewer 2: But if we're to follow existing examples, this should be named submenu instead.
```suggestion
        $this->addApiDocSubMenu($menu);
```

        return $menu;
    }

>     private function addApiDocMenu(ItemInterface $menu): void

Reviewer 2: Personally, I would prefer some other example external submenu/url name than api doc, something in line with books and library from test app and other example snippets.

WDYT @loic425 @stlgaits

Reviewer: Yes good point to try to remain in userland - possibly an external link to the bookstore's website / blog or to something related to the books ecosystem like Goodreads  new releases  for instance (e.g. https://www.goodreads.com/book/popular_by_date/2025/5?ref=nav_brws_newrels) ?

Author: I changed it to a link to the website homepage. This is not in the test app, but we've got something similar in `Sylius/Sylius`

>     private function addApiDocMenu(ItemInterface $menu): void
>     {
>         $library = $menu
>             ->addChild('api_doc', [
>                 'route' => 'api_doc',
>             ])
>             ->setLabel('app.ui.api_doc')
>             ->setLabelAttribute('icon', 'tabler:api')
>             ->setLinkAttribute('target', '_blank') // Opens the link in a new tab
>         ;
>     }

Reviewer 2: ```suggestion
    private function addApiDocSubMenu(ItemInterface $menu): void
    {
        $apiDoc = $menu
            ->addChild('api_doc', [
                'route' => 'api_doc',
            ])
            ->setLabel('app.ui.api_doc')
            ->setLabelAttribute('icon', 'tabler:api')
            ->setLinkAttribute('target', '_blank') // Opens the link in a new tab
        ;
    }
```

}
```

> 

Reviewer: ```suggestion
{% endcode %}

```

> >>>>>>> Stashed changes

Reviewer 2: ```suggestion
```

Author: :see_no_evil:
